shremote.py

```diff
@@ -159,14 +159,18 @@ def format(cls, st, **kwargs):
         if cls.instance_ is None:
             raise Exception("Config instance not instantiated")
 
-        # TODO: Regex match, in case string contains \{
-        while '{' in st:
+        # TODO: check if open bracket is matched
+        while len([m.start() for m in re.finditer('[^\\\\]\{', st)]) > 0:
```

Comment:
Does this (and the other places) work? I'm surprised it doesn't complain about `\{` not being a valid escape sequence in a normal python string...
I think this should be `r'[^\\]\{'` (marked as a raw string with r, so that it doesn't try to escape anything in the string itself.